docs/general/ad-filtering/create-own-filters.md

```diff
@@ -2141,6 +2142,99 @@ Rules with `$replace` modifier are supported by AdGuard for Windows, Mac, and An
 
 :::
 
+#### **`urltransform`** {#urltransform-modifier}
+
+`$urltransform` rules allow to modify the request url by replacing text matched by regular expression.
+
+**Features**
+
+- `$urltransform` rules apply to any request url text.
+- `$urltransform` rules can also **modify the query part** of the url.
+- `$urltransform` will not be applied if original url get blocked by other rules.
+- `$urltransform` will be applied before `$removeparam` rule
+
+`$urltransform` value can be empty in the case of exception rules.
+
+**Multiple rules matching a single request**
+
+In case if multiple `$urltransform` rules match a single request, we will apply each of them. **The order is defined alphabetically.**
+
+**Syntax**
+
+`$urltransform` syntax is similar to replacement with regular expressions [in Perl](http://perldoc.perl.org/perlrequick.html#Search-and-replace).
+
+```text
+urltransform = "/" regexp "/" replacement "/" modifiers
+```
+
+- **`regexp`** — a regular expression.
+- **`replacement`** — a string that will be used to replace the string corresponding to `regexp`.
+- **`modifiers`** — a regular expression flags. For example, `i` — insensitive search, or `s` — single-line mode.
+
+In the `$urltransform` value, two characters must be escaped: comma `,` and dollar sign `$`. Use backslash `\` for it. For example, an escaped comma looks like this: `\,`.
+
+**Examples**
+
+```adblock
+||example.org^$urltransform=/(pref\/).*\/(suf)/\$1\$2/i
+```
+
+There are three parts in this rule:
+
+- `regexp` — `(pref\/).*\/(suf)`;
+- `replacement` — `\$1\$2` where `$` is escaped;
+- `modifiers` — `i` for insensitive search.
+
+**Multiple `$urltransform` rules**
+
+1. `||example.org^$urltransform=/X/Y/`
+2. `||example.org^$urltransform=/Z/Y/`
+3. `@@||example.org/page/*$urltransform=/Z/Y/`
+
+- Both rule 1 and 2 will be applied to all requests sent to `example.org`.
+- Rule 2 is disabled for requests matching `||example.org/page/`, **but rule 1 still works!**
+
+**Re-matching rules after transforming url**
+
+If the `$urltransform` rule had beed applied to a request, it will be re-matched against the url filter again, because url text has been changed and we might want to block it.
```

Comment:
What is a "url filter" here?

Do you mean that ALL rules will be re-evaluated?

Reply:
`url filter` is responsible for matching requests against rules. `$urltransform` is the most priority rule to be applied to a request. And if it is applied, the url can become completely different, thus we need to build a new rules matching against this new url because there is likely a different set of rules for it. None of the rules are re-evaluated because there are no rules applied (except `$urltransform`) at this point.

Reply:
Should we remove the mention about "url filter" to not confuse the user?

Reply:
It'd be better to not use internal naming here and just say that all rules will be reevaluated against the new URL.